exporters/zipkin/zipkin_test.go

```diff
@@ -375,3 +376,40 @@ func TestLogrFormatting(t *testing.T) {
 	got := buf.String()
 	assert.Equal(t, want, got)
 }
+
+func TestWithHeaders(t *testing.T) {
+	headers := map[string]string{
+		"name1": "value1",
+		"name2": "value2",
+		"host":  "example",
+	}
+
+	exp, err := New("", WithHeaders(headers))
+	require.NoError(t, err)
+
+	want := headers
+	got := exp.headers
+	assert.Equal(t, want, got)
+
+	ch := make(chan *http.Request, 1)
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		ch <- r
+		w.WriteHeader(http.StatusOK)
+	}
+
+	srv := httptest.NewServer(http.HandlerFunc(handler))
+	defer srv.Close()
+
+	e := &Exporter{
+		url:     srv.URL,
+		client:  srv.Client(),
+		headers: headers,
+	}
+
+	_ = e.ExportSpans(context.Background(), tracetest.SpanStubs{}.Snapshots())
+	req := <-ch
```

Comment:
ExportSpans runs synchronously. So once the method has finished running, the request should be populated. Hence, we shouldn't need a channel.

```suggestion
	var req *http.Request
	handler := func(w http.ResponseWriter, r *http.Request) {
		req = r
		w.WriteHeader(http.StatusOK)
	}

	srv := httptest.NewServer(http.HandlerFunc(handler))
	defer srv.Close()

	e := &Exporter{
		url:     srv.URL,
		client:  srv.Client(),
		headers: headers,
	}

	_ = e.ExportSpans(context.Background(), tracetest.SpanStubs{}.Snapshots())
```

Reply:
Yes, I tried this way but request comes out to be `nil` as per this. Hence, used channel to read value of request.

Reply:
yeah, got it.. I forgot to add a span, will add it.

Reply:
Done the changes, please check.